src/pages/index.js

```diff
@@ -1,103 +1,193 @@
 
 const previewModal = document.querySelector("#preview-modal");
 const modalImage = previewModal.querySelector(".modal__image");
-const previewCaption = previewModal.querySelector(".modal__caption");
 
 // Buttons
 
 const profileEditBtn = document.querySelector("#profile-edit-button");
 const addNewCardBtn = document.querySelector("#add-card-button");
-const closeBtns = document.querySelectorAll(".modal__close-button");
 
-// FUNCTIONS
+// API'S
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "250daeea-0b63-48f9-a4b3-e529116433c4",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    newCardSection.setItems(cards);
+    newCardSection.renderItems();
+  })
+  .catch((err) => {
+    alert(`${err} Failed to get cards.`);
+  });
+
+api
+  .getUserInfo()
+  .then((info) => {
+    profileUserInfo.setUserInfo({
+      name: info.name,
+      description: info.about,
+    });
+    profileUserInfo.setUserAvatar(info.avatar);
+  })
+  .catch((err) => {
+    alert(`${err} Failed to get user info.`);
+  });
+
+// API FUNCTIONS
 
 function handleProfileEditSubmit(data) {
-  profileUserInfo.setUserInfo(data);
+  profileEditForm.setLoading(true);
+  api
+    .updateUserInfo(data)
+    .then((result) => {
+      profileUserInfo.setUserInfo(data);
+      profileEditForm.close();
+    })
+    .catch((err) => {
+      alert(`${err} Failed to change user info.`);
+    })
+    .finally(() => {
+      profileEditForm.setLoading(false);
+    });
 }
 
 function handleCardSubmit({ title: name, subtitle: link }) {
-  newCardSection.addItem({ name, link });
-  addImageForm.close();
+  addImageForm.setLoading(true);
+  api
+    .addCard({ name, link })
+    .then((card) => {
+      newCardSection.addItem({ name, link });
+      addImageForm.close();
+    })
+    .catch((err) => {
+      alert(`${err} Failed to add card.`);
+    })
+    .finally(() => {
+      addImageForm.setLoading(false);
+    });
+}
+
+function handleAvatarSubmit(input) {
+  avatarEditForm.setLoading(true);
+  api
+    .changeAvatar(input.link)
+    .then((result) => {
+      profileUserInfo.setUserAvatar(result.avatar);
+      avatarEditForm.setLoading(false);
+    })
+    .catch((err) => {
+      alert(`${err} Failed to change avatar.`);
+    });
 }
 
+function handleAddLike(card) {
+  if (card.isLiked) {
+    return api
+      .removeLike(card._id)
+      .then((res) => {
+        card.setIsLiked(res.isLiked);
+      })
+      .catch((err) => {
+        alert(`${err} Failed to add/remove like.`);
+      });
+  } else {
+    return api
+      .setLike(card._id)
+      .then((res) => {
+        card.setIsLiked(res.isLiked);
+      })
+      .catch((err) => {
+        alert(`${err} Failed to add/remove like.`);
+      });
+  }
+}
+
+function handleDeleteClick(card) {
+  confirmDeletePopup.open();
+  confirmDeletePopup.setSubmitAction(() => {
+    api
```

Comment:
Needs correction: you shoiuld also notify the user about loading process by changing the confirm button text

Reply:
So like the other submit buttons, do you mean add that same feature? For example when clicking the Yes button to delete a card it should say "Saving..." or "Deleting"?

Reply:
Yes, you're right